tests/candev/main.c

```diff
@@ -100,12 +104,14 @@ static int _receive(int argc, char **argv)
         puts("Reading from Rxbuf...");
         isrpipe_read(&rxbuf, buf, 4);       /* can-id */
         can_id = ((uint32_t)buf[0] << 24) |
-                ((uint32_t)buf[1] << 16) |
-                ((uint32_t)buf[2] << 8) |
-                ((uint32_t)buf[3]);
+                 ((uint32_t)buf[1] << 16) |
+                 ((uint32_t)buf[2] << 8) |
+                 ((uint32_t)buf[3]);
         isrpipe_read(&rxbuf, buf, 1);       /* can-dlc */
         can_dlc = buf[0];
-        isrpipe_read(&rxbuf, buf, can_dlc); /* data */
+        if (can_dlc > 0) {
+            isrpipe_read(&rxbuf, buf, can_dlc); /* data */
+        }
```

Comment:
Why this change?

Reply:
This was a bug in the candev app that was discovered while testing this PR. It does not exclusively apply to MCP2515 but to all possible can-devices you can use in the candev test app. 

This fix prevents the test-app from entering a deadlock when CAN messages without a payload are received (which is allowed in CAN). However... if I remember correctly, I found it pretty strange that this fix was necessary. Imho it is a weird design choice of the isrpipe system in RIOT. But changing that system is very much out of scope for this PR :) 

Since this fix isn't exclusively tied to mcp2515, maybe it makes sense to put it in a separate commit? (A separate PR is probably a bit too much for something this small...)

Reply:
ok!